quickcomm/views.py

```diff
@@ -159,15 +162,35 @@ def post_like(request, post_id, author_id):
     # post = get_object_or_404(Post, pk=post_id)
 
     if request.user.is_authenticated:
-        author = Author.objects.all().get(user=request.user)
-        like_obj, created_obj = Like.objects.get_or_create(post=post, author=author)
-        if not created_obj:
-            like_obj.delete()
-        like_key = f"post_like_{post_id}"
-        request.session[like_key] = not created_obj
-
+        if request.method == 'POST':
+            author = Author.objects.all().get(user=request.user)
+            like_obj, created_obj = Like.objects.get_or_create(post=post, author=author)
+            if not created_obj:
+                post.likes.remove(request.user)
```

Comment:
So the issue you were having with the NoReverseMatch is actually from having the ability to unlike posts. When you remove a Like like this, a new Inbox item is created (can't quite figure out why? when are these false objects actually added to the Inbox?) but it no longer has a Like attached to it, so it tries to find the Like with id 0, and obviously fails. The easy fix is to disallow unliking of posts and comments, as I don't think it's strictly required.

This could, however, lead to another issue down the line when deleting a Post, as obviously we don't want to rewrite history and remove the fact that a Post once existed from everyone else's Inbox, but that means that all of the hyperlinks within the Inbox item will not be able to retrieve anything as the Object they were once attached to no longer exists.